extension/src/test/Xliff/XliffCache.test.ts

```diff
@@ -0,0 +1,65 @@
+import { xliffCache } from "../../Xliff/XLIFFCache";
+import * as path from "path";
+import { Xliff } from "../../Xliff/XLIFFDocument";
+import * as assert from "assert";
+
+suite("XliffCache Tests", () => {
+  const cachedFilePath = path.join(
+    __dirname,
+    "../../../",
+    "src/test/resources/NAB_AL_Tools.sv-SE.xlf"
+  );
+  test("xliffCache.add()", function () {
+    xliffCache.update(
+      cachedFilePath,
+      Xliff.fromFileSync(cachedFilePath).toString()
+    );
+    assert.ok(
+      xliffCache.isCached(path.basename(cachedFilePath)),
+      "Expected file to be cached."
+    );
+    assert.ok(xliffCache.isEnabled, "Expected XliffCache to be enabled.");
+  });
+
+  test("xliffCache.isCached()", function () {
+    assert.ok(
+      xliffCache.isCached(cachedFilePath),
+      "Expected file to be cached."
+    );
+  });
+
+  test("xliffCache.get()", function () {
+    const cachedXlf = xliffCache.get(cachedFilePath);
+    assert.ok(
+      cachedXlf.transunit.length > 0,
+      "No transunits found in cached Xliff"
+    );
+  });
+
+  test("xliffCache.delete()", function () {
+    assert.ok(xliffCache.delete(cachedFilePath));
+    assert.strictEqual(
+      xliffCache.isCached(cachedFilePath),
+      false,
+      "File should not be cached."
+    );
+  });
+
+  test("xliffCache.size", function () {
+    assert.strictEqual(
+      xliffCache.size,
+      1,
```

Comment:
hmm... What am I missing...? How can this be `1` when you delete that from cache in the test above?

Reply:
Magic ;) It's due to the event test for `onDidChangeTextDocument`. Maybe I should throw in a `clear()` in that test.

Reply:
meh! 🤯

I think we should separate tests where possible, so adding a `xliffCache.clear()` should be best practices and having dependency only where really needed. (not for lazy reasons 😅)